esp32c3-hal/examples/adc_cal.rs

```diff
@@ -0,0 +1,77 @@
+//! Connect a potentiometer to PIN2 and see the read values change when
+//! rotating the shaft. Alternatively you could also connect the PIN to GND or
+//! 3V3 to see the maximum and minimum raw values read.
+
+#![no_std]
+#![no_main]
+
+use esp32c3_hal::{
+    adc::{AdcCalBasic, AdcCalCurve, AdcCalLine, AdcConfig, Attenuation, ADC, ADC1},
+    clock::ClockControl,
+    gpio::IO,
+    peripherals::Peripherals,
+    prelude::*,
+    timer::TimerGroup,
+    Delay,
+    Rtc,
+};
+use esp_backtrace as _;
+use esp_println::println;
+
+#[entry]
+fn main() -> ! {
+    let peripherals = Peripherals::take();
+    let mut system = peripherals.SYSTEM.split();
+    let clocks = ClockControl::boot_defaults(system.clock_control).freeze();
+
+    // Disable the watchdog timers. For the ESP32-C3, this includes the Super WDT,
+    // the RTC WDT, and the TIMG WDTs.
+    let mut rtc = Rtc::new(peripherals.RTC_CNTL);
+    let timer_group0 = TimerGroup::new(
+        peripherals.TIMG0,
+        &clocks,
+        &mut system.peripheral_clock_control,
+    );
+    let mut wdt0 = timer_group0.wdt;
+    let timer_group1 = TimerGroup::new(
+        peripherals.TIMG1,
+        &clocks,
+        &mut system.peripheral_clock_control,
+    );
+    let mut wdt1 = timer_group1.wdt;
+
+    rtc.swd.disable();
+    rtc.rwdt.disable();
+    wdt0.disable();
+    wdt1.disable();
+
+    let io = IO::new(peripherals.GPIO, peripherals.IO_MUX);
+
+    // Create ADC instances
+    let analog = peripherals.APB_SARADC.split();
+
+    let mut adc1_config = AdcConfig::new();
+
+    // type AdcCal = ();
+    // type AdcCal = (AdcCalBasic<ADC1>,);
+    // type AdcCal = (AdcCalBasic<ADC1>, AdcCalLine<ADC1>);
+    type AdcCal = (AdcCalBasic<ADC1>, AdcCalLine<ADC1>, AdcCalCurve<ADC1>);
```

Comment:
Why do you have to set 3 different calibration types? What happens if I just define `type AdcCal = (AdcCalCurve<ADC1>);`?